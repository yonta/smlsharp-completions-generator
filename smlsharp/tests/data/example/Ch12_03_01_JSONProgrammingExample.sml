val J = "[{\"name\":\"Joe\", \"age\":21, \"grade\":1.1},"
                 ^ "{\"name\":\"Sue\", \"age\":31, \"grade\":2.0},"
                 ^ "{\"name\":\"Bob\", \"age\":41, \"grade\":3.9}]";
fun getNames l = map #name l;
val j = JSON.import J;
val vl = _json j as {name:string, age:int, grade:real} list;
val nl = getNames vl;
