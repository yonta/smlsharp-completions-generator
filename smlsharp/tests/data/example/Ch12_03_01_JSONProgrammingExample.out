val J =
  "[{\"name\":\"Joe\", \"age\":21, \"grade\":1.1},{\"name\":\"Sue\", \"age\":31, \"grade\":2.0},{\"name\":\"Bob\", \"age\":41, \"grade\":3.9}]" : string

val getNames = fn : ['a#{name: 'b}, 'b. 'a list -> 'b list]

val j = _ : JSON.void JSON.dyn

val vl =
  [
   {age = 21, grade = 1.1, name = "Joe"},
   {age = 31, grade = 2.0, name = "Sue"},
   {age = 41, grade = 3.9, name = "Bob"}
  ] : {age: int, grade: real, name: string} list

val nl = ["Joe", "Sue", "Bob"] : string list
